iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -459,7 +516,7 @@ private void OnDesiredPropertyReceived(Twin twin, string correlationId, TwinColl
         public async Task SendTwinMessageAsync(AmqpTwinMessageType amqpTwinMessageType, string correlationId, TwinCollection reportedProperties, TimeSpan timeout)
         {
             if (Logging.IsEnabled) Logging.Enter(this, timeout, $"{nameof(SendTwinMessageAsync)}");
-
+            await EnableTwinLinksAsync(timeout).ConfigureAwait(false);
```

Comment:
This should not be called unless the links are gone.
There is overhead in calling an async so there should be a check before calling.

Reply:
Need a lock to handle race condition here. By design